app/rpc/rpchandlers/submit_block.go

```diff
@@ -34,6 +34,22 @@ func HandleSubmitBlock(context *rpccontext.Context, _ *router.Router, request ap
 		}, nil
 	}
 
+	if !context.Config.AllowSubmittingNonDAABlocks {
+		virtualDAAScore, err := context.Domain.Consensus().GetVirtualDAAScore()
+		if err != nil {
+			return nil, err
+		}
+		// A simple heuristic check which signals that the mined block is out of date
+		// and should not be accepted unless user explicitly requests
+		if int(virtualDAAScore-domainBlock.Header.DAAScore()) > context.Config.NetParams().DifficultyAdjustmentWindowSize {
```

Comment:
@someone235 I'm not sure if I should test the case `virtualDAAScore < domainBlock.Header.DAAScore()` and what does it even mean to be in such state. Currently this will overflow and the condition will be true, hence block rejected